poetry/core/json/__init__.py

```diff
@@ -31,9 +31,7 @@ def validate_object(obj: dict, schema_name: str) -> List[str]:
     for error in validation_errors:
         message = error.message
         if error.path:
-            message = "[{}] {}".format(
-                ".".join(str(x) for x in error.absolute_path), message
-            )
+            message = f"[{'.'.join(str(x) for x in error.absolute_path)}] {message}"
```

Comment:
i think this is less clear than the original. consider splitting into two lines
```suggestion
            path = '.'.join(str(x) for x in error.absolute_path)
            message = f"[{path}] {message}"
```